h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -398,21 +407,27 @@ to a map (possibly the metadata map) -
                         storeLock.unlock();
                     }
                 }
+                
+                lastCommitTime = getTimeSinceCreation();
+
+                scrubMetaMap();
+
+                // setAutoCommitDelay starts the thread, but only if
+                // the parameter is different from the old value
+                int delay = DataUtils.getConfigParam(config, "autoCommitDelay", 1000);
+                setAutoCommitDelay(delay);
+                
+                success = true;
             } catch (IllegalStateException e) {
                 panic(e);
             } finally {
+                if (!success && !fileStoreIsProvided) {
+                    closeImmediately();
+                }
```

Comment:
why not just expand scope of the catch above and use panic() instead? This way it would report exception to interested listener and record it (look at MVStore.store()). Maybe panic() itself should be amended with "if (!fileStoreIsProvided)" logic?

Reply:
OK.